src/cryptonote_core/loki_name_system.h

```diff
@@ -47,30 +66,36 @@ enum struct burn_type
   custom,
 };
 
-inline std::ostream &operator<<(std::ostream &os, mapping_type type)
-{
-  switch(type)
-  {
-    case mapping_type::lokinet: os << "lokinet"; break;
-    case mapping_type::session: os << "session"; break;
-    case mapping_type::wallet:  os << "wallet"; break;
-    default: assert(false);     os << "xx_unhandled_type"; break;
-  }
-  return os;
-}
-
+burn_type      mapping_type_to_burn_type(mapping_type in);
 constexpr bool mapping_type_allowed(uint8_t hf_version, mapping_type type) { return type == mapping_type::session; }
-burn_type    mapping_type_to_burn_type(mapping_type in);
-uint64_t     burn_requirement_in_atomic_loki(uint8_t hf_version, burn_type type);
+uint64_t       burn_requirement_in_atomic_loki(uint8_t hf_version, burn_type type);
+
+
 sqlite3     *init_loki_name_system(char const *file_path);
 uint64_t     lokinet_expiry_blocks(cryptonote::network_type nettype, uint64_t *renew_window = nullptr);
 crypto::hash tx_extra_signature_hash(epee::span<const uint8_t> blob, crypto::hash const &prev_txid);
 bool         validate_lns_name(mapping_type type, std::string const &name, std::string *reason = nullptr);
 
-// blob: if set, validate_lns_value will convert the value into the binary format suitable for storing into the LNS DB.
-bool         validate_lns_value(cryptonote::network_type nettype, mapping_type type, std::string const &value, lns_value *blob = nullptr, std::string *reason = nullptr);
-bool         validate_lns_value_binary(mapping_type type, std::string const &value, std::string *reason = nullptr);
-bool         validate_mapping_type(std::string const &type, lns::mapping_type *mapping_type, std::string *reason);
+// Validate a human readable mapping value representation in 'value' and write the binary form into 'blob'.
+// value: if type is session, 64 character hex string of an ed25519 public key
```

Comment:
Shouldn't the session length be 66 (i.e. with the leading 05)?

Reply:
Yes updated